rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -575,6 +576,27 @@ bool microphone_device::has_data() const
 	return mic_registered && mic_opened && mic_started && (rbuf_raw.has_data() || rbuf_dsp.has_data());
 }
 
+f32 microphone_device::calculate_energy_level()
+{
+	const auto& buffer = device_type == microphone_handler::real_singstar ? ::at32(devices, 0).buf : temp_buf;
+
+	const be_t<s16>* samples = utils::bless<const be_t<s16>>(buffer.data());
+	const size_t num_samples = buffer.size() / sizeof(s16);
+
+	f32 sum_squares = 0.0f;
+	for (usz i = 0; i < num_samples; i++)
+	{
+		const f32 normalized_sample = static_cast<f32>(samples[i]) / std::numeric_limits<s16>::max();
```

Comment:
This would return a number higher than 1 for INT16_MIN, maybe divide by (0 - INT16_MIN)